server/data_common/dataset_metadata.py

```diff
@@ -85,30 +85,27 @@ def get_dataset_metadata_for_explorer_location(dataset_explorer_location: str, a
                 raise TombstoneError(message=msg, collection_id=collection_id, dataset_id=dataset_id)
             return dataset_metadata
 
-    server_config = app_config.server_config
-    dataset_metadata = {
+
+    current_app.logger.log(logging.INFO, f"Dataset not found by Data Portal: {explorer_url_path}. "
+                                         "Falling back to deriving S3 location from request URL.")
+
+    s3_uri = infer_dataset_s3_uri(
+        server_config=app_config.server_config, dataset_root=dataset_root, dataset_id=dataset_id)
+
+    return {
         "collection_id": None,
         "collection_visibility": None,
         "dataset_id": None,
-        "s3_uri": None,
+        "s3_uri": s3_uri,
         "tombstoned": False,
     }
-    dataset_metadata["s3_uri"] = extrapolate_dataset_location_from_config(
-        server_config=server_config, dataset_explorer_location=dataset_explorer_location
-    )
-    if dataset_metadata["s3_uri"] is None:
-        current_app.logger.log(logging.INFO, f"Dataset not found: {dataset_explorer_location}")
-        raise DatasetNotFoundError(f"Dataset location not found for {dataset_explorer_location}")
-
-    return dataset_metadata
```

Comment:
Above is refactoring of `get_dataset_metadata_for_explorer_location()`, renaming to `get_dataset_metadata()` and also now taking the separate `dataset_root` and dataset_id` params for clarity.  Improved logging messages.